mmv1/third_party/tgc_next/test/utils.go

```diff
@@ -19,3 +24,93 @@ func writeJSONFile(filename string, data interface{}) error {
 	}
 	return nil
 }
+
+const (
+	defaultOrganization = "529579013760"
+	defaultProject      = "ci-test-project-nightly-beta"
+)
+
+func terraformWorkflow(t *testing.T, dir, name string) {
+	terraformInit(t, "terraform", dir)
+	terraformPlan(t, "terraform", dir, name+".tfplan")
+	payload := terraformShow(t, "terraform", dir, name+".tfplan")
+	saveFile(t, dir, name+".tfplan.json", payload)
+}
+
+func terraformInit(t *testing.T, executable, dir string) {
+	terraformExec(t, executable, dir, "init", "-input=false")
+}
+
+func terraformPlan(t *testing.T, executable, dir, tfplan string) {
+	terraformExec(t, executable, dir, "plan", "-input=false", "-refresh=false", "-out", tfplan)
+}
+
+func terraformShow(t *testing.T, executable, dir, tfplan string) []byte {
+	return terraformExec(t, executable, dir, "show", "--json", tfplan)
+}
+
+func terraformExec(t *testing.T, executable, dir string, args ...string) []byte {
+	cmd := exec.Command(executable, args...)
+	cmd.Env = []string{
+		"HOME=" + filepath.Join(dir, "fakehome"),
+		"GOOGLE_PROJECT=" + defaultProject,
+		"GOOGLE_FOLDER=" + "",
+		"GOOGLE_ORG=" + defaultOrganization,
+		"GOOGLE_OAUTH_ACCESS_TOKEN=fake-token", // GOOGLE_OAUTH_ACCESS_TOKEN is required so terraform plan does not require the google authentication cert
+	}
+	if os.Getenv("TF_CLI_CONFIG_FILE") != "" {
+		cmd.Env = append(cmd.Env, "TF_CLI_CONFIG_FILE="+os.Getenv("TF_CLI_CONFIG_FILE"))
+	}
+	cmd.Dir = dir
+	wantError := false
+	payload, _ := run(t, cmd, wantError)
+	return payload
+}
+
+func saveFile(t *testing.T, dir, filename string, payload []byte) {
+	fullpath := filepath.Join(dir, filename)
+	f, err := os.Create(fullpath)
+	if err != nil {
+		t.Fatalf("error while creating file %s, error %v", fullpath, err)
+	}
+	_, err = f.Write(payload)
+	if err != nil {
+		t.Fatalf("error while writing to file %s, error %v", fullpath, err)
+	}
+}
+
+// run a command and call t.Fatal on non-zero exit.
+func run(t *testing.T, cmd *exec.Cmd, wantError bool) ([]byte, []byte) {
+	var stderr, stdout bytes.Buffer
+	cmd.Stderr, cmd.Stdout = &stderr, &stdout
+	err := cmd.Run()
+	if gotError := (err != nil); gotError != wantError {
+		t.Fatalf("running %s: \nerror=%v \nstderr=%s \nstdout=%s", cmdToString(cmd), err, stderr.String(), stdout.String())
+	}
+	// Print env, stdout and stderr if verbose flag is used.
+	if len(cmd.Env) != 0 {
+		t.Logf("=== Environment Variable of %s ===", cmdToString(cmd))
+		t.Log(strings.Join(cmd.Env, "\n"))
+	}
+	if stdout.String() != "" {
+		t.Logf("=== STDOUT of %s ===", cmdToString(cmd))
+		t.Log(stdout.String())
+	}
+	if stderr.String() != "" {
+		t.Logf("=== STDERR of %s ===", cmdToString(cmd))
+		t.Log(stderr.String())
+	}
+	return stdout.Bytes(), stderr.Bytes()
+}
+
+// cmdToString clones the logic of https://golang.org/pkg/os/exec/#Cmd.String.
```

Comment:
why not use the stdlib function in that case?

Reply:
Good point. Changing to Cmd.String() method instead.